dart/lib/src/scope.dart

```diff
@@ -142,6 +142,65 @@ class Scope {
   /// Removes an extra from the Scope
   void removeExtra(String key) => _extra.remove(key);
 
+  SentryEvent applyToEvent(SentryEvent event, dynamic hint) {
+    if (event.transaction == null) {
+      event = event.copyWith(transaction: transaction);
+    }
+
+    // set the user context if none is set.
+    if (event.userContext == null) {
+      event = event.copyWith(userContext: user);
+    }
+
+    // set the scope fingerprint if none is set.
+    if (event.fingerprint == null) {
+      event = event.copyWith(fingerprint: fingerprint);
+    }
+
+    // set the scope breadcrumbs if none is set.
+    if (event.breadcrumbs == null) {
+      event = event.copyWith(breadcrumbs: breadcrumbs);
+    }
```

Comment:
on list and maps we could check if they are not null and empty, so we also save allocation

Reply:
✅